pegasus_wrapper/resource_request.py

```diff
@@ -142,24 +164,25 @@ def apply_to_job(self, job: Job, *, job_name: str) -> None:
         if not self.partition:
             raise RuntimeError("A partition to run on must be specified.")
 
+        if self.partition.walltime < self.job_time_in_minutes:
```

Comment:
Instead of checking at each direct or `from_params` instantiation of `SlurmResourceRequest`, I catch the error when a user tries to apply their configuration to a job.